pkgs/os-specific/linux/unscd/default.nix

```diff
@@ -0,0 +1,59 @@
+{ fetchurl, fetchpatch, stdenv, systemd, lib }:
+
+stdenv.mkDerivation rec {
+  pname = "unscd";
+  version = "0.54";
+
+  src = fetchurl {
+    url = "https://busybox.net/~vda/${pname}/nscd-${version}.c";
+    sha256 = "0iv4iwgs3sjnqnwd7dpcw6s7i4ar9q89vgsms32clx14fdqjrqch";
+  };
+
+  unpackPhase = ''
+    cp $src nscd.c
+    chmod u+w nscd.c
+  '';
+
+  patches = [
+    ./remove-old-socket.diff
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/change_invalidate_request_info_output";
+      sha256 = "17whakazpisiq9nnw3zybaf7v3lqkww7n6jkx0igxv4z2r3mby6l";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/support_large_numbers_in_config";
+      sha256 = "0jrqb4cwclwirpqfb6cvnmiff3sm2jhxnjwxa7h0wx78sg0y3bpp";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/no_debug_on_invalidate";
+      sha256 = "0znwzb522zgikb0mm7awzpvvmy0wf5z7l3jgjlkdpgj0scxgz86w";
+    })
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/u/${pname}/${version}-1/debian/patches/notify_systemd_about_successful_startup";
+      sha256 = "1ipwmbfwm65yisy74nig9960vxpjx683l3skgxfgssfx1jb9z2mc";
+    })
+  ];
+
+  buildInputs = [ systemd ];
+
+  buildPhase = ''
+    gcc -Wall \
```

Comment:
This should probably use the NIX_CFLAGS_COMPILE or similar instead of overwriting buildPhase.

Reply:
There is no (versioned) Makefile, does the default buildPhase detect .c files in the src and compile them even when there is no Makefile?

Reply:
I don't know but I think we should definitely add the default flags.

Reply:
So... There is an upstream Makefile. I do not use it, because the url is not versioned, so changes to the Makefile would break the hash. But the command I run in buildPhase is exactly what _would_ be run with the upstream Makefile. It overrides CFLAGS specifically to this.